### ISSUE TITLE
Critical: TensorFlow Version Compatibility Issues with Python 3.9 and Keras Dependencies

### ISSUE THREAD

**Issue by aaron-seq** [aaronseq12/BPpredictiontool#5]
## Problem Description

The project has serious compatibility issues between the specified TensorFlow version (2.13.0) and Python 3.9 runtime environment that will prevent the application from running properly.

### 1. TensorFlow Version Compatibility
- **Current Configuration**: TensorFlow 2.13.0 with Python 3.9
- **Problem**: TensorFlow 2.13.0 requires Python 3.8-3.11, but optimal compatibility is with Python 3.8-3.10
- **Impact**: Installation failures and runtime incompatibilities in production environments

### 2. Keras Import Issues
The advanced_lstm_model.py file uses deprecated import patterns:
```python
from tensorflow.keras.layers import (
    Input, LSTM, Dense, Dropout, BatchNormalization,
    TimeDistributed, Attention, MultiHeadAttention,
    LayerNormalization, Add, GlobalAveragePooling1D
)
```

**Problem**: Direct TensorFlow Keras imports may cause compatibility issues with newer TensorFlow versions where some components have been moved or deprecated.

### 3. Model Architecture Compatibility
The AttentionBloodPressureModel uses:
- MultiHeadAttention layers that may have parameter changes between TensorFlow versions
- Custom attention mechanisms that may not be compatible with TensorFlow 2.13.0's implementation
- Layer normalization and dropout patterns that could be affected by version differences

### 4. Production Deployment Issues
The Docker configuration specifies Python 3.9-slim, but the TensorFlow/Keras combination may have performance and stability issues in containerized environments.

### Immediate Actions Required

1. **Update TensorFlow Configuration**:
   ```
   tensorflow==2.15.0
   ```
   (Latest stable version with better Python 3.9 support)

2. **Update Keras Imports**:
   ```python
   import tensorflow as tf
   from tensorflow.keras import layers, Model
   # Use tf.keras.layers.MultiHeadAttention instead of direct import
   ```

3. **Version Alignment**:
   - Update requirements.txt with compatible versions
   - Test the model loading and training pipeline
   - Verify attention mechanism compatibility

4. **Docker Environment Update**:
   - Consider using Python 3.10 slim image for better TensorFlow support
   - Update Dockerfile with proper GPU support if needed

### Testing Required
- Verify model loading from the specified paths
- Test the attention mechanism functionality
- Validate the LSTM architecture with new TensorFlow version
- Check preprocessing pipeline compatibility

## Impact Assessment
- **Severity**: Critical
- **Component**: Backend/ML Model
- **Deployment**: Affects all deployment targets (Railway, Vercel, Docker)
- **User Experience**: Application will fail to start or produce incorrect predictions

## Files Affected
- `backend/requirements.txt`
- `backend/models/advanced_lstm_model.py` 
- `Dockerfile`
- `backend/main.py` (model loading functions)

This issue blocks the application from running and must be resolved before any other functionality can be tested or used.
